pkg/distributor/distributor.go

```diff
@@ -483,18 +497,35 @@ func (d *Distributor) validateSeries(ts cortexpb.PreallocTimeseries, userID stri
 		return emptyPreallocSeries, err
 	}
 
-	samples := make([]cortexpb.Sample, 0, len(ts.Samples))
-	for _, s := range ts.Samples {
-		if err := validation.ValidateSample(d.limits, userID, ts.Labels, s); err != nil {
-			return emptyPreallocSeries, err
+	// Don't alloc a new slice unnecessarily when input is empty
+	samples := ts.Samples
```

Comment:
Would this functionally be any different? If `ts.Samples` is 0 length we just have another reference to it with `samples := ts.Samples`, with `var samples []cortexpb.Sample` aren't we allocating an additional slice header (ptr, len, cap)?

Reply:
My thought was that assigning `nil` for Samples when only Exemplars present (for example) would prevent returning the Samples slice to the pool, but it is actually ok, since the original `req` still contains the original slices.  Nil slice `var samples []cortexpb.Sample` should be fine if we prefer that for readability

Reply:
> Nil slice var samples []cortexpb.Sample should be fine if we prefer that for readability

There's no logic difference, but I think it's just easier to read.